mm2src/coins/solana.rs

```diff
@@ -226,35 +230,32 @@ pub struct SolanaFeeDetails {
 }
 
 async fn withdraw_base_coin_impl(coin: SolanaCoin, req: WithdrawRequest) -> WithdrawResult {
-    let (to_send, my_balance) = coin.check_sufficient_balance(req.max, req.amount).await?;
-    let base_balance = coin.base_coin_balance().compat().await?;
-    let hash = coin.rpc().get_latest_blockhash().await?;
+    let (hash, fees) = coin.estimate_withdraw_fees().await?;
+    let (to_send, my_balance, sol_required) = coin.check_sufficient_balance(req.max, req.amount.clone(), fees).await?;
+    let lamports_to_send = if req.max {
+        sol_to_lamports(&my_balance)? - sol_to_lamports(&sol_required)?
+    } else {
+        sol_to_lamports(&req.amount)?
```

Comment:
In the case when `req.max == false` we can first generate transaction, and then estimate the fee for that tx.
What do you think of this instruction?
* If `max == true`
  * Generate a dummy transaction, calculate required fees
  * Calculate total amount to send
  * Generate desired transaction with `amount = my_balance - fees`
* else `max == true`
  * Generate desired transaction with `amount = req.amount`

It makes sense if we can neglect the cost of  calling the `solana_sdk::system_transaction::transfer` function **only**.
It means that the function does not take a long time or does not consume a lot of resources.

Reply:
> In the case when `req.max == false` we can first generate transaction, and then estimate the fee for that tx. What do you think of this instruction?
> 
> * If `max == true`
>   
>   * Generate a dummy transaction, calculate required fees
>   * Calculate total amount to send
>   * Generate desired transaction with `amount = my_balance - fees`
> * else `max == true`
>   
>   * Generate desired transaction with `amount = req.amount`
> 
> It makes sense if we can neglect the cost of calling the `solana_sdk::system_transaction::transfer` function **only**. It means that the function does not take a long time or does not consume a lot of resources.

I solved everything else, as mentioned before `transfer` and `estimate_fee_for_message` have almost no delay and are not costly so I think we can keep this approach.